Sources/Frisbee/Interactors/ResultGenerator.swift

```diff
@@ -10,8 +10,14 @@ final class ResultGenerator<T: Decodable> {
 
     func generate(data: Data?, error: Error?) -> Result<T> {
         guard let data = data else {
-            let frisbeeError = FrisbeeError(error ?? FrisbeeError.noData)
```

Comment:
I was trying to abstract error construction inside the `FrisbeError` constructor.  What you think? If you agree, could be something like this:
```swift
public enum FrisbeeError: Error {

    // ...

    init(_ error: Error?) {
        guard let frisbeeError = error as? FrisbeeError else {
            switch error {
                 case .some(URLError.cancelled):
                     self = .requestCancelled
                 case .some(let error):
                     self = .other(localizedDescription: error.localizedDescription)
                 case .none:
                     return .noData
             }
        }
        self = frisbeeError
    }
}
```

With this, it isn't necessary to to change the `ResultGeneratorTests`. Just add an test for `FrisbeeError`

Reply:
Maybe with my suggestion, the best approach is to replace `guard/let` by `if/let`